app/src/main/kotlin/com/flxrs/dankchat/data/repo/UserDisplayRepository.kt

```diff
@@ -0,0 +1,90 @@
+package com.flxrs.dankchat.data.repo
+
+import com.flxrs.dankchat.data.database.UserDisplayDao
+import com.flxrs.dankchat.data.database.UserDisplayEntity
+import com.flxrs.dankchat.data.twitch.message.*
+import com.flxrs.dankchat.di.ApplicationScope
+import com.flxrs.dankchat.preferences.userdisplay.UserDisplayDto
+import com.flxrs.dankchat.preferences.userdisplay.UserDisplayDto.Companion.toDto
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import javax.inject.Inject
+import javax.inject.Singleton
+
+
+@Singleton
+class UserDisplayRepository @Inject constructor(
+    private val userDisplayDao: UserDisplayDao,
+    @ApplicationScope val coroutineScope: CoroutineScope,
+) {
+    suspend fun getAllUserDisplays(): List<UserDisplayDto> = userDisplayDao.getAll().map { it.toDto() }
+
+    private val userDisplays = userDisplayDao.getUserDisplaysFlow().map { users -> users.map { it.toDto() } }.stateIn(
+        coroutineScope,
+        SharingStarted.Eagerly,
+        emptyList()
+    )
+
+    /** add (if ID == 0) or update (ID != 0) user displays */
+    suspend fun addUserDisplays(userDisplays: List<UserDisplayDto>) {
+        userDisplayDao.insertAll(userDisplays.map { it.toEntity() })
+    }
+
+    suspend fun deleteByIds(ids: List<Int>): Int {
+        return userDisplayDao.deleteAll(ids.map { UserDisplayEntity.makeDummy(it) })
```

Comment:
This is not needed if you go with the approach I use for highlights/ignores. I just map the `Items` used by the ui back to the corresponding entities.